lib/internal/modules/ts/typescript.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const {
+  ArrayPrototypeSome,
+  String,
+  StringPrototypeEndsWith,
+} = primordials;
+const { transformSync } = require('internal/deps/swc/wasm');
+const {
+  stringify,
+} = require('internal/modules/helpers');
+const internalFsBinding = internalBinding('fs');
+const {
+  ERR_UNSUPPORTED_DIR_IMPORT,
+} = require('internal/errors').codes;
+const { fileURLToPath, URL } = require('internal/url');
+const { toNamespacedPath } = require('path');
+
+async function load(url, context, nextLoad) {
+  const isTS = context.format === 'typescript';
+  if (isTS) { context = { __proto__: context, format: 'module' }; }
+  const next = await nextLoad(url, context);
+  if (isTS) {
+    const { code } = transformSync(stringify(next.source), {
+      swcrc: false,
+      jsc: {
+        parser: {
+          syntax: 'typescript',
+        },
+        target: 'esnext',
+      },
+    });
+    return {
+      __proto__: null,
+      format: 'module',
+      source: code,
+    };
+  }
+  return next;
+}
+
+async function resolve(specifier, context, nextResolve) {
+  // Check if the file has a known extension
+  const extensions = ['.ts', '.cjs', '.mjs', '.js'];
+  const hasKnownExtension = ArrayPrototypeSome(extensions, (ext) => StringPrototypeEndsWith(specifier, ext));
```

Comment:
That's incorrect, if e.g. you have `import "viz.js"`, it's referencing the name of a package named `viz.js`, but it should not resolve to a relative import. You should first call `nextResolve` and act on the resulting `url`. In general, I think it's a bad idea to try to resolve magic-extension, let's do like Deno in the first PR (i.e. only support import that contains the `.ts` extension)

Reply:
the issue I have faced if there isnt a `.ts` extension, for example `import foo from "foo"`, what I'm trying to do is check if `foo.ts` exists and use it, otherwise go to the default loader if it has a known extension